src/trajectory_tracking.h

```diff
@@ -0,0 +1,62 @@
+
+#ifndef __TRACY__
+#define __TRACY__
+
+#include <stdint.h>
+
+#define DEFAULT_PARAM_FANCY_GREEK_C (0.7f)
+#define DEFAULT_PARAM_G (60.0f)
+
+/**
+ * output of the linear controller
+ *
+ */
+typedef struct {
+    float rule1;
+    float rule2;
+} feedback_rule_t;
+
+/**
+ * represents a tracking error: reference_pos - current_pos
+ *
+ */
+typedef struct {
+    float x_error;
+    float y_error;
+    float theta_error;
+} tracking_error_t;
+
+/**
+ * velocity of reference in polar robot coordinates
+ *
+ */
+typedef struct {
+    float radial_velocity;
+    float angular_velocity;
+} reference_velocity_t;
+
+/**
+ * compute the linear controller output
+ *
+ * @param current_error error between where you want to be and where you are 
+ *                      at this timestep
+ * @param reference_velocity velocity you want to be at currently
+ * @param output pointer to memory to store result
+ *
+ * @return 0 if everything went fine, or something else on failure
```

Comment:
You don't have to check for NULL pointers. The problem is, that if the function can fail we have to handle the error case in the caller. It is a lot easier to have math functions that never fail